### PR TITLE
Test3 verb for multiple TFM

### PR DESCRIPTION
**Pull request by Faizan2304**
1) Added support to run test from test3 verb for the project targeting multiple TFM
2) Added test to address above change
3) Crossgen were creating problem for full CLR built assemblies. Removed All those assembly from Crossgen list.
4) Removed the line of code which were deleting Extensions folder under sdk folder as this folder contain trx logger dlls

### CODE REVIEW

**Comment by Faizan2304** [comment]
@dotnet-bot test this please

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" />
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>net46;netcoreapp1.0</TargetFrameworks>
+    <OutputPath>bin\$(Configuration)</OutputPath>
```
Comment (livarcocc) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: I don't think you need output path.

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp1.0'">
```
Comment (livarcocc) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: Do you really need this?

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    <DefineConstants>NETCOREAPP;$(DefineConstants)</DefineConstants>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net46'">
+    <DefineConstants>DESKTOP;$(DefineConstants)</DefineConstants>
```
Comment (livarcocc) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: And this.

File: build/Microsoft.DotNet.Cli.Compile.targets [hunk 4]
```diff
@@ -216,8 +219,6 @@
       <Copy SourceFiles="@(MSBuildTargetsToCopy)"
             DestinationFiles="@(MSBuildTargetsToCopy->'$(SdkOutputDirectory)/%(RecursiveDir)%(Filename)%(Extension)')" />
 
-      <RemoveDir Directories="$(SdkOutputDirectory)/Extensions" />
```
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Compile.targets]: Have you seen and what is inside that folder besides your files? And how big that is? I think the right thing here is to still remove all the other files and just leave yours.

File: test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs [hunk 5]
```diff
@@ -28,12 +30,39 @@ public void TestsFromAGivenProjectShouldRunWithExpectedOutput()
                 .Pass();
 
             // Call test3
-            CommandResult result = new Test3Command().WithWorkingDirectory(testProjectDirectory).ExecuteWithCapturedOutput("/p:TargetFramework=netcoreapp1.0");
+            CommandResult result = new Test3Command().WithWorkingDirectory(testProjectDirectory).ExecuteWithCapturedOutput("--framework netcoreapp1.0");
```
Comment (livarcocc) [inline test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs]: NIT: please, keep the line length to at most 120 characters.

File: test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs [hunk 6]
```diff
@@ -28,12 +30,39 @@ public void TestsFromAGivenProjectShouldRunWithExpectedOutput()
...
+
+            // Restore project VSTestDotNetCoreProject
+            new Restore3Command()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should()
+                .Pass();
+
+            string configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+            string expectedError = Path.Combine(testProjectDirectory, "bin", configuration, "netcoreapp1.0", "VSTestDotNetCoreProject.dll");
```
Comment (livarcocc) [inline test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs]: NIT: 120 characters.

File: test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs [hunk 7]
```diff
@@ -28,12 +30,39 @@ public void TestsFromAGivenProjectShouldRunWithExpectedOutput()
...
+            expectedError = "The test source file " + "\""+ expectedError + "\"" + " provided was not found.";
+
+            // Call test3
+            CommandResult result = new Test3Command().WithWorkingDirectory(testProjectDirectory).ExecuteWithCapturedOutput("--noBuild");
```
Comment (livarcocc) [inline test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs]: NIT: 120 characters.

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj (hunk 1 above)
Comment (Faizan2304) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: Yes we don't need it. I verified without output path and its working fine.

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj (hunk 2 above)
Comment (Faizan2304) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: No, I don't need. It is working fine without these two line. Removed in next commit.

File: TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj (hunk 3 above)
Comment (Faizan2304) [inline TestAssets/TestProjects/VSTestDesktopAndNetCoreApp/VSTestDesktopAndNetCoreApp.csproj]: Yes I need this. On the basis of this tag I am compiling different test for different TFM.

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 4 above)
Comment (Faizan2304) [inline build/Microsoft.DotNet.Cli.Compile.targets]: There were no files other than mine.

File: test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs (hunk 5 above)
Comment (Faizan2304) [inline test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs]: Fixed.

File: test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs (hunk 6 above)
Comment (Faizan2304) [inline test/dotnet-test3.Tests/GivenDotnetTest3BuildsAndRunsTestfromCsproj.cs]: Fixed

File: build/Microsoft.DotNet.Cli.Compile.targets (hunk 4 above)
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Compile.targets]: All right then.

**Comment by Faizan2304** [comment]
@dotnet-bot test this please

**Comment by Faizan2304** [comment]
@dotnet-bot test this please

**Pull request merged**